src/styles/themer/withTheme.js

```diff
@@ -0,0 +1,41 @@
+// Higher order component that force updates component on themer change
+// and passes down theme through props
+import React, { Component } from 'react'
+import themer from './index'
+import { themePropTypes } from './utils'
+
+function withTheme(InnerComponent) {
+  class Wrapped extends Component {
+    static displayName = `withTheme(${InnerComponent.displayName})`
+
+    static propTypes = {
+      snacksTheme: themePropTypes
+    }
+
+    componentDidMount() {
+      this.unsubscribe = themer.subscribe(this.onThemeChange)
+    }
+
+    componentWillUnmount() {
+      this.unsubscribe()
+    }
+
+    onThemeChange() {
+      this.forceUpdate()
+    }
+
+    render() {
+      return (
+        <InnerComponent
+          ref={(node) => this.wrapped = node}
+          snacksTheme={themer.themeConfig}
```

Comment:
We might want to put this after we spread `{...this.props}`, someone may accidentally pass `snacksTheme` and overwrite the prop.

Reply:
Yeah, I couldn't decide if that should be allowed or disallowed. I went with allowing it to be overwritten for more flexibility. What do you think @wfro?

Reply:
Don't feel too strongly either way but I'm find with it being overridable, most won't ever interact with the `snacksTheme` prop directly anyway, and so doesn't seem like a major footgun

Reply:
But since this is all the HOC does, seems weird to allow the only prop it passes down to be overridden

Reply:
Hm, that's a really good point. 😓  Why override it if you're using the HOC?

Ok, going to change the ordering.